### DISCUSSION TITLE
[Feature] Backup restart on network switch

### DISCUSSION THREAD

**Discussion by Chiogros** [immich-app/immich#2060] [Feature Request]
### The feature

Hi,

When switching from data to Wi-Fi while backuping assets, the stream is cutt off and the server throws an exception:
```
immich_server              | [Nest] 1  - 03/23/2023, 4:17:48 PM   ERROR [ExpressAdapter] Premature close
immich_server              | Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
immich_server              |     at new NodeError (node:internal/errors:372:5)
immich_server              |     at ServerResponse.onclose (node:internal/streams/end-of-stream:142:30)
immich_server              |     at ServerResponse.emit (node:events:539:35)
immich_server              |     at Socket.onServerResponseClose (node:_http_server:236:23)
immich_server              |     at Socket.emit (node:events:539:35)
immich_server              |     at TCP.<anonymous> (node:net:709:12)
```

It would be great to handle such an exception, so that the mobile app can display an error message or restart the stopped asset upload.

Thanks for your work,
Alexandre.

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by jrasm91** [comment]
Can you confirm that the mobile app backup process does not retry the asset automatically? I believe it should.

This server error is actually just a notice that a read stream has been destroyed when `pipeline` was used, so it was unexpected that is was destroyed before it ended.

**Comment by Chiogros** [comment]
Sincerely sorry,
I was fooled by my reverse proxy.
